### PR TITLE
PD-349  Sessions Namespace and PD-351 Two_Factor Namespace

### PR DESCRIPTION
**Pull request by tonyriv3**

### CODE REVIEW

File: content/SCALE/SCALECLIReference/Auth/CLISessions.md [hunk 1]
```diff
@@ -12,4 +12,225 @@ tags:
...
+Enter the command, then press <kbd>Enter</kbd>.
+The command returns a table with the ID, current session state, origin such as an IP or socket information, credentials, and creation date and time.
+Enter the command, then press <kbd>Enter</kbd>.
+The command returns a table with the ID, current session state, origin such as an IP or socket information, credentials, and creation date and time.
+
+#### Usage
+
+From the CLI Auth namespace prompt, enter:
+
+`sessions`
```
Comment (micjohnson777) [inline content/SCALE/SCALECLIReference/Auth/CLISessions.md:38]: I think we are standardizing on the full namespace path as the example. This is not incorrect but we should use the same string start point.
```suggestion
`auth sessions`
```

File: content/SCALE/SCALECLIReference/Auth/CLISessions.md [hunk 2]
```diff
@@ -12,4 +12,225 @@ tags:
...
+```
+{{< /expand >}}
+{{< /expand >}}
+
+### Current Command
+
+The `current` command displays whether or not the TrueNAS sessions in the `sessions` list are current. 
+
+{{< expand "Displaying if sessions are current" "v" >}}
+The `current` command has no required options. It returns a table showing which sessions are current. The rows of the table are relative to the rows returned from the `sessions` command.
```
Comment (micjohnson777) [inline content/SCALE/SCALECLIReference/Auth/CLISessions.md:84]:
```suggestion
The `current` command has no required options. 
Enter the command, then press <kbd>Enter</kbd>.
It returns a table showing which sessions are current. 
The rows of the table are relative to the rows returned from the `sessions` command.
```

File: content/SCALE/SCALECLIReference/Auth/CLISessions.md [hunk 3]
```diff
@@ -12,4 +12,225 @@ tags:
...
+{{< /expand >}}
+{{< /expand >}}
+
+### ID Command
+
+The `id` command displays the IDs of all TrueNAS sessions since boot. 
+
+{{< expand "Identifying sessions" "v" >}}
+The `id` command has no required options. It returns a table of recent and active session IDs.
+
```
Comment (micjohnson777) [inline content/SCALE/SCALECLIReference/Auth/CLISessions.md:59]: Not following the template in this section.
Add the line about entering the command and the result.

File: content/SCALE/SCALECLIReference/Auth/CLISessions.md [hunk 4]
```diff
@@ -12,4 +12,225 @@ tags:
...
+{{< /expand >}}
+{{< /expand >}}
+
+### Current Command
+
+The `current` command displays whether or not the TrueNAS sessions in the `sessions` list are current. 
+
+{{< expand "Displaying if sessions are current" "v" >}}
+The `current` command has no required options. It returns a table showing which sessions are current. The rows of the table are relative to the rows returned from the `sessions` command.
+
```
Comment (micjohnson777) [inline content/SCALE/SCALECLIReference/Auth/CLISessions.md:85]: Where do these possible states come in? Is it part of the table information returned? If it is, you can just add this in a results line, as in, "Returns true if the session is current, false if not current."

File: content/SCALE/SCALECLIReference/Auth/CLISessions.md [hunk 5]
```diff
@@ -12,4 +12,225 @@ tags:
...
+```
+{{< /expand >}}
+{{< /expand >}}
+
+### Internal Command
+
+The `internal` command displays whether sessions in the `sessions` list are internally created (via the web UI) or not (via SSH).
+
+{{< expand "Displaying if sessions are internal" "v" >}}
+The `internal` command has no required options. It returns a table showing if sessions are internal or not. The rows of the table are relative to the rows returned from the `sessions` command.
```
Comment (micjohnson777) [inline content/SCALE/SCALECLIReference/Auth/CLISessions.md:118]: Include the line instructing them on how to enter the command.
```suggestion
The `internal` command has no required options. 
Enter the command, then press <kbd>Enter</kbd>.
The command returns a table showing if sessions are internal or not. 
The rows of the table are relative to the rows returned from the `sessions` command.
```

File: content/SCALE/SCALECLIReference/Auth/CLISessions.md [hunk 6]
```diff
@@ -12,4 +12,225 @@ tags:
...
+```
+{{< /expand >}}
+{{< /expand >}}
+
+### Origin Command
+
+The `origin` command displays the login origin of the sessions in the `sessions` list.
+
+{{< expand "Displaying the login origin of sessions" "v" >}}
+The `origin` command has no required options. It returns a table showing the login origin of each session. The rows of the table are relative to the rows returned from the `sessions` command.
```
Comment (micjohnson777) [inline content/SCALE/SCALECLIReference/Auth/CLISessions.md:152]: same comment above. Include the "enter the command, then press Enter line per the template.

File: content/SCALE/SCALECLIReference/Auth/CLISessions.md [hunk 7]
```diff
@@ -12,4 +12,225 @@ tags:
...
+```
+{{< /expand >}}
+{{< /expand >}}
+
+### Credentials Command
+
+The `credentials` command displays the credentials used to authenticate each session in the `sessions` list.
+
+{{< expand "Displaying how each session authenticated" "v" >}}
+The `credentials` command has no required options. It returns a table showing the credentials used to authenticate each session. The rows of the table are relative to the rows returned from the `sessions` command.
```
Comment (micjohnson777) [inline content/SCALE/SCALECLIReference/Auth/CLISessions.md:178]: same comment. Missing the enter command instruction. Also, tie in the table with the possible states so the user makes this connection better. For example, See the possible states table below for more information on what the credentials use to authenticate each session, or something similar.

**Review by micjohnson777 — CHANGES_REQUESTED** [review]

File: content/SCALE/SCALECLIReference/Auth/CLISessions.md [hunk 8]
```diff
@@ -12,4 +12,260 @@ tags:
...
+
+Enter the command, then press <kbd>Enter</kbd>.
+
+The command returns a table of recent and active session IDs.
+
+#### Usage
+
+From the CLI Auth namespace prompt, enter:
+
+`sessions id`
```
Comment (micjohnson777) [inline content/SCALE/SCALECLIReference/Auth/CLISessions.md:72]:
```suggestion
` auth sessions id`
```

File: content/SCALE/SCALECLIReference/Auth/CLISessions.md [hunk 9]
```diff
@@ -12,4 +12,260 @@ tags:
...
+
+Enter the command, then press <kbd>Enter</kbd>.
+
+The command returns a table with the ID, current session state, origin such as an IP or socket information, credentials, and creation date and time.
+
+#### Usage
+
+From the CLI Auth namespace prompt, enter:
+
+`sessions`
```
Comment (micjohnson777) [inline content/SCALE/SCALECLIReference/Auth/CLISessions.md:39]:
```suggestion
`auth sessions`
```

File: content/SCALE/SCALECLIReference/Auth/CLISessions.md [hunk 10]
```diff
@@ -12,4 +12,260 @@ tags:
...
+
+### Sessions Command
+
+The `sessions` command returns a table that displays each session id, whether or not each session is current and internal, each session origin, and the credentials and credential data used to log into each session.
+
+{{< expand "Displaying all sessions since system boot" "v" >}}
+#### Description
+
+The `sessions` command does not require entering properties or arguments.
+
```
Comment (micjohnson777) [inline content/SCALE/SCALECLIReference/Auth/CLISessions.md:30]: you can delete this empty line here in in line 32. And really all the commands. You can make these a paragraph.

File: content/SCALE/SCALECLIReference/Auth/CLISessions.md [hunk 11]
```diff
@@ -12,4 +12,260 @@ tags:
...
+Enter the command, then press <kbd>Enter</kbd>.
+
+The command returns a table showing which sessions are current (true) or not (false). The rows of the table are relative to the rows returned from the `sessions` command.
+
+#### Usage
+
+From the CLI Auth namespace prompt, enter:
+
+`sessions current`
```
Comment (micjohnson777) [inline content/SCALE/SCALECLIReference/Auth/CLISessions.md:105]:
```suggestion
`auth sessions current`
```

File: content/SCALE/SCALECLIReference/Auth/CLISessions.md [hunk 12]
```diff
@@ -12,4 +12,260 @@ tags:
...
+|----------|-------------|
+| `true`   | Indicates the session is internal (created via the web UI). |
+| `false`  | Indicates the session is not internal (created via SSH). |
+{{< /truetable >}}
+
+#### Usage
+
+From the CLI Auth namespace prompt, enter:
+
+`sessions internal`
```
Comment (micjohnson777) [inline content/SCALE/SCALECLIReference/Auth/CLISessions.md:146]:
```suggestion
`auth sessions internal`
```

File: content/SCALE/SCALECLIReference/Auth/CLISessions.md [hunk 13]
```diff
@@ -12,4 +12,260 @@ tags:
...
+
+Enter the command, then press <kbd>Enter</kbd>.
+
+The command returns a table showing the login origin of each session. The rows of the table are relative to the rows returned from the `sessions` command.
+
+#### Usage
+
+From the CLI Auth namespace prompt, enter:
+
+`sessions origin`
```
Comment (micjohnson777) [inline content/SCALE/SCALECLIReference/Auth/CLISessions.md:179]:
```suggestion
`auth sessions origin`
```

File: content/SCALE/SCALECLIReference/Auth/CLISessions.md [hunk 14]
```diff
@@ -12,4 +12,260 @@ tags:
...
+| `LOGIN_PASSWORD`  | Indicates the session used SSH password authentication.                 |
+| `API_KEY`         | Indicates the session used API key authentication.                      |
+| `TOKEN`           | Indicates the session used token authentication.                        |
+{{< /truetable >}}
+
+#### Usage
+
+From the CLI Auth namespace prompt, enter:
+
+`sessions credentials`
```
Comment (micjohnson777) [inline content/SCALE/SCALECLIReference/Auth/CLISessions.md:223]:
```suggestion
`auth sessions credentials`
```

File: content/SCALE/SCALECLIReference/Auth/CLISessions.md [hunk 15]
```diff
@@ -12,4 +12,260 @@ tags:
...
+
+Enter the command, then press <kbd>Enter</kbd>.
+
+The command returns a table showing the creation date and time of each session. The rows of the table are relative to the rows returned from the `sessions` command.
+
+#### Usage
+
+From the CLI Auth namespace prompt, enter:
+
+`sessions created_at`
```
Comment (micjohnson777) [inline content/SCALE/SCALECLIReference/Auth/CLISessions.md:256]:
```suggestion
` auth sessions created_at`
```

**Review by micjohnson777 — CHANGES_REQUESTED** [review]
This review covers the sessions namespace article.
You can remove the single line paragraphs and make the three a single paragraph. 
command examples start with the parent namespace.
Otherwise excellent!

File: content/SCALE/SCALECLIReference/Auth/CLITwoFactor.md [hunk 16]
```diff
@@ -12,4 +12,243 @@ tags:
...
+The command returns nothing when successful, and returns an error when you enter an invalid integer.
+
+#### Usage
+
+From the CLI prompt, enter:
+
+<code>auth two_factor update interval=<i>number</i></code>
+
+Where:
+* *number* is the number (in seconds) a OTP will last before expiring. Must be between 5 and 999999999999999999.
```
Comment (micjohnson777) [inline content/SCALE/SCALECLIReference/Auth/CLITwoFactor.md:194]: Don't use will
```suggestion
* *number* is the number (in seconds) a OTP lasts before expiring. Must be between 5 and 999999999999999999.
```

File: content/SCALE/SCALECLIReference/Auth/CLITwoFactor.md [hunk 17]
```diff
@@ -12,4 +12,243 @@ tags:
...
+{{< /expand >}}
+
+### Verify Command
+
+The `verify` command verifies whether or not a password is authenticated.
+
+{{< expand "Renewing the 2FA Secret">}}
+
+#### Description
+The `verify` command requires the `token` property.
```
Comment (micjohnson777) [inline content/SCALE/SCALECLIReference/Auth/CLITwoFactor.md:233]: if this requires the token property, then the next line should tell them how to enter the property argument (we have standard text for this in the template .

File: content/SCALE/SCALECLIReference/Auth/CLITwoFactor.md [hunk 18]
```diff
@@ -12,4 +12,243 @@ tags:
...
+Enter the command, then press <kbd>Enter</kbd>.
```
Comment (micjohnson777) [inline content/SCALE/SCALECLIReference/Auth/CLITwoFactor.md:234]:
```suggestion
Enter the command string, then press <kbd>Enter</kbd>.
```

**Review by micjohnson777 — DISMISSED** [review]
Just a few questions and template points. You do such good work!

**Review by Mrt134 — APPROVED** [review]

**Pull request merged**